File: SpectrumWeb/Models/AdAndSb.cs
using System;
using System.Collections.Generic;

namespace SpectrumWeb.Models;

public partial class AdAndSb
{
    public string PkRecordId { get; set; } = null!;

    public string PkTransactionId { get; set; } = null!;

    public int? Revision { get; set; }

    public int? Number { get; set; }

    public int? ModNmbr { get; set; }

    public string? AircraftType { get; set; }

    public string? PartNumber { get; set; }

    public string? TypeCode { get; set; }

    public bool? ComplyYN { get; set; }

    public string? VendorNumber { get; set; }

    public bool? AlertYN { get; set; }

    public DateOnly? DateEntered { get; set; }

    public DateOnly? DateComplete { get; set; }

    public string? Description { get; set; }

    public string? Resolution { get; set; }

    public DateOnly? ComplianceDate { get; set; }

    public string? AtaChap { get; set; }

    public double? ThresholdHours { get; set; }

    public int? ThresholdCycles { get; set; }

    public int? ThresholdDays { get; set; }

    public int? ThresholdWeeks { get; set; }

    public bool? BridgeToInspec { get; set; }

    public bool? InclInInspec { get; set; }

    public bool? OneTimeInspec { get; set; }

    public string? ComplyChain1 { get; set; }

    public string? ComplyChain2 { get; set; }

    public string? ComplyChain3 { get; set; }

    public DateOnly? DateIssued { get; set; }

    public double? PerformEveryHour { get; set; }

    public short? PerformEveryCycle { get; set; }

    public short? PerformEveryDay { get; set; }

    public short? PerformEveryWeek { get; set; }

    public string? ApplyToSerNmbr { get; set; }

    public string? ThruSerNmbr { get; set; }

    public bool? Terminate { get; set; }

    public string? TerminateReason { get; set; }

    public DateOnly? DateTerminated { get; set; }
}

File: SpectrumWeb/wwwroot/Local/ad_and_sb.js
class ADandSBClass {
    PkRecordId;
    TransactId;
    Number;
    ModNmbr;
    AircraftType;
    Adact1;
    Adact2;
    PartNumber;
    TypeCode;
    ComplyYN;
    VendorNumber;
    AlertYN;
    DateEntered;
    DateComp;
    Description;
    Resolution;
    TailNmbrs;
    ComplianceDate;
    ACModNmbr;
    AtaChap;
    ThresholdHrs;
    ThresholdCyc;
    ThresholdWks;
    BridgeToInspec;
    OneTimeInspec;
    RevisionNmbr;
    AlsoComplyNmbr1;
    AlsoComplyNmbr2;
    AlsoComplyNmbr3;
    AlsoComplyNmbr4;
    DateIssued;
    PerformEveryHr;
    PerformEveryCy;
    PerformEveryWk;
    ApplyToSerNmbr;
    ThruSerNmbr;
    Terminate;
    TerminateReason;
    DateTerminated;

    constructor(
        pkRecordId,
        transactId,
        number,
        modNmbr,
        aircraftType,
        adact1,
        adact2,
        partNumber,
        typeCode,
        complyYN,
        vendorNumber,
        alertYN,
        dateEntered,
        dateComp,
        description,
        resolution,
        tailNmbrs,
        complianceDate,
        aCModNmbr,
        ataChap,
        thresholdHrs,
        thresholdCyc,
        thresholdWks,
        bridgeToInspec,
        oneTimeInspec,
        revisionNmbr,
        alsoComplyNmbr1,
        alsoComplyNmbr2,
        alsoComplyNmbr3,
        alsoComplyNmbr4,
        dateIssued,
        performEveryHr,
        performEveryCy,
        performEveryWk,
        applyToSerNmbr,
        thruSerNmbr,
        terminate,
        terminateReason,
        dateTerminated
    ) {
        this.PkRecordId = pkRecordId;
        this.TransactId = transactId;
        this.Number = number;
        this.ModNmbr = modNmbr;
        this.AircraftType = aircraftType;
        this.Adact1 = adact1;
        this.Adact2 = adact2;
        this.PartNumber = partNumber;
        this.TypeCode = typeCode;
        this.ComplyYN = formatBooleanAsYN(complyYN);
        this.VendorNumber = vendorNumber;
        this.AlertYN = alertYN;
        this.DateEntered = formatDateTimeAsDate(dateEntered);
        this.DateComp = formatDateTimeAsDate(dateComp);
        this.Description = description;
        this.Resolution = resolution;
        this.TailNmbrs = tailNmbrs;
        this.ComplianceDate = formatDateTimeAsDate(complianceDate);
        this.ACModNmbr = aCModNmbr;
        this.AtaChap = ataChap;
        this.ThresholdHrs = thresholdHrs;
        this.ThresholdCyc = thresholdCyc;
        this.ThresholdWks = thresholdWks;
        this.BridgeToInspec = bridgeToInspec;
        this.OneTimeInspec = formatBooleanAsYN(oneTimeInspec);
        this.RevisionNmbr = revisionNmbr;
        this.AlsoComplyNmbr1 = alsoComplyNmbr1;
        this.AlsoComplyNmbr2 = alsoComplyNmbr2;
        this.AlsoComplyNmbr3 = alsoComplyNmbr3;
        this.AlsoComplyNmbr4 = alsoComplyNmbr3;
        this.DateIssued = formatDateTimeAsDate(dateIssued);
        this.PerformEveryHr = performEveryHr;
        this.PerformEveryCy = performEveryCy;
        this.PerformEveryWk = performEveryWk;
        this.ApplyToSerNmbr = applyToSerNmbr;
        this.ThruSerNmbr = thruSerNmbr;
        this.Terminate = formatBooleanAsYN(terminate);
        this.TerminateReason = terminateReason;
        this.DateTerminated = formatDateTimeAsDate(dateTerminated);
    }
}